src/middlewared/middlewared/plugins/unscheduled_reboot_alert.py

```diff
@@ -0,0 +1,116 @@
+            return
+
+        gc = await middleware.call('datastore.config', 'network.globalconfiguration')
+        hostname = f"{gc['gc_hostname']}.{gc['gc_domain']}"
+        now = datetime.now().strftime("%c")
+
+        # If the watchdog alert file exists,
+        # then we can assume that carp-state-change-hook.py
+        # panic'ed the box by design via a watchdog countdown.
+        # Let's alert the end user why we did this
+        if os.path.exists(WATCHDOG_ALERT_FILE) and not os.path.exists(FENCED_ALERT_FILE):
+            await middleware.call("mail.send", {
+                "subject": f"{hostname}: Failover event",
+                "text": textwrap.dedent(f"""\
+                    {hostname} had a failover event.
+                    The system was rebooted to ensure a proper failover occurred.
+                    The operating system successfully came back online at {now}.
+                """),
+            })
+
+        # If the fenced alert file exists,
+        # then we can assume that fenced panic'ed the box by design.
+        # Let's alert the end user why we did this
+        elif os.path.exists(FENCED_ALERT_FILE) and not os.path.exists(WATCHDOG_ALERT_FILE):
+            await middleware.call("mail.send", {
+                "subject": f"{hostname}: Failover event",
+                "text": textwrap.dedent(f"""\
+                    {hostname} had a failover event.
+                    The system was rebooted because persistent SCSI reservations were lost and/or cleared.
+                    The operating system successfully came back online at {now}.
+                """),
+
+            })
+
+        # If both files exist, then something weird happened.
+        # Get the modify time of the files, and send an email accordingly
+        elif os.path.exists(WATCHDOG_ALERT_FILE) and os.path.exists(FENCED_ALERT_FILE):
+            watchdog_time = os.path.getmtime(WATCHDOG_ALERT_FILE)
+            fenced_time = os.path.getmtime(FENCED_ALERT_FILE)
+            if (watchdog_time > fenced_time):
+                await middleware.call("mail.send", {
+                    "subject": f"{hostname}: Failover event",
+                    "text": textwrap.dedent(f"""\
+                        {hostname} had a failover event.
+                        The system was rebooted to ensure a proper failover occurred.
+                        The operating system successfully came back online at {now}.
+                    """),
+
+                })
+
+            else:
+                await middleware.call("mail.send", {
+                    "subject": f"{hostname}: Failover event",
+                    "text": textwrap.dedent(f"""\
+                        {hostname} had a failover event.
+                        The system was rebooted because persistent SCSI reservations were lost and/or cleared.
+                        The operating system successfully came back online at {now}.
+                    """),
+                })
```

Comment:
Not critical, but I would reorganize this code to not duplicate messages.  Something like `if (exist(A) and (!exist (B) or mtime(A) > mtime(B)) ... elif (exist(B)) ... else ...`